src/main/java/com/bookshop/controller/BookController.java

```diff
@@ -30,26 +32,26 @@ public List<BookDto> getAll() {
 
     @ResponseStatus(HttpStatus.OK)
     @GetMapping("/{id}")
-    public BookDto getBookById(@PathVariable Long id) {
+    public BookDto getBookById(@PathVariable @Positive Long id) {
```

Comment:
@PathVariable @Positive Long id
In order to be able to perform such validation you need to annotate your controller class with Validated annotation